kernel/pci/src/lib.rs

```diff
@@ -129,7 +157,18 @@ pub struct PciBus {
 
 /// Scans all PCI Buses (brute force iteration) to enumerate PCI Devices on each bus.
 /// Initializes structures containing this information. 
-fn scan_pci() -> Vec<PciBus> {
+fn scan_pci() -> Result<Vec<PciBus>, &'static str> {
+    #[cfg(target_arch = "aarch64")]
+    PCI_CONFIG_SPACE.lock().try_call_once(|| {
+        let config = BOARD_CONFIG.pci_ecam;
+        let flags = memory::PteFlags::new().valid(true).writable(true);
```

Comment:
do we need to map this as  device memory?

Reply:
Indeed! since the PCI controller can be in a north bridge, which is a separate chip behind CPU caches.
I missed this, thank you for spotting it.